src/components/RestockDialog.vue

```diff
@@ -0,0 +1,320 @@
+    if (input !== null) {
+        return true
+    }
+
+    return 'The amount must be specified.'
+}
+
+/**
+ * Helper function -- this is a rule -- to check if the input is a natural number.
+ *
+ * @param input - The input string to check.
+ * @returns True if the input is a natural number; an error message otherwise.
+ */
+function inputIsNaturalNumber(input: string): boolean | string {
+    if (!inputIsNotNull(input)) {
+        return false
+    }
+
+    if (/^\d+$/.test(input) && parseInt(input) > 0) {
+        return true
+    }
+
+    return 'The input must be a natural number, e.g. 5.'
+}
+
+/**
+ * Helper function to check if the input is smaller than the maximum batch size.
+ *
+ * @param input - The input string to check.
+ * @returns True if the input is smaller than the maximum batch size; an error message otherwise.
+ */
+function inputIsSmallerThanMaximumBatchSize(input: string): boolean | string {
+    if (!inputIsNaturalNumber(input)) {
+        return false
+    }
+
+    if (props.maximumBatchSize === undefined) {
+        return true
+    }
+
+    const number = parseInt(input)
+
+    if (number <= props.maximumBatchSize && number > 0) {
+        return true
+    }
+
+    return (
+        'The amount may not exceed the specified maximum batch size of: ' +
+        props.maximumBatchSize
+    )
+}
+
+/**
+ * Ref for the preselected product variant ID.
+ */
+const selectedProductVariantId = ref<string | undefined>(
+    props.preselectedProductVariantId
+)
+
+/**
+ * Ref for the amount input.
+ */
+const amount = ref<string | null>(null)
+
+/**
+ * Computed property indicating whether there are changes in the component's state.
+ */
+const hasChanges = computed(() => {
+    return (
+        selectedProductVariantId.value !== props.preselectedProductVariantId ||
+        amount.value !== null
+    )
+})
+
+/**
+ * Resets the view model to its initial state.
+ * @returns Whether or not the view model was reset.
+ */
+function resetViewModel(): boolean {
+    if (hasChanges.value) {
+        selectedProductVariantId.value = props.preselectedProductVariantId
+
+        amount.value = null
+
+        return true
+    }
+
+    return false
+}
+
+/**
+ * Resets the view model if necessary and emits the event that the dialog should be closed.
+ */
+function cancel() {
+    resetViewModel()
+
+    emit('close')
+}
+
+/**
+ * Computed property indicating whether restocking can be requested.
+ */
+const canRestock = computed(() => {
+    return (
+        selectedProductVariantId.value !== undefined &&
+        amount.value !== null &&
+        inputIsNaturalNumber(amount.value) &&
+        inputIsSmallerThanMaximumBatchSize(amount.value)
+    )
+})
+
+/**
+ * Requests the restocking of the selected product variant.
+ */
+async function restock() {
+    if (activeUserRoleIsEitherAdminOrEmployee.value && canRestock) {
+        const q = await pushErrorNotificationIfNecessary(() => {
```

Comment:
what does `q` mean? was this a temporary variable name?

Reply:
Yes, forgot to clean it up. [There you go](https://github.com/MiSArch/frontend/pull/37/commits/a3ae4b7e7badf322bdec788162fa1feec2682eaa)